src/js/components/Button/StyledButton.js

```diff
@@ -96,7 +96,9 @@ const StyledButton = styled.button`
   background: transparent;
   overflow: visible;
   text-transform: none;
-
+  &:focus {
```

Comment:
I thinks this needs a check for `focusIndicator` to not show the focus styling if `focusIndicator` is false.